flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnalignerTest.java

```diff
@@ -512,6 +512,123 @@ public void testConcurrentProcessBarrierAndNotifyBarrierReceived() throws Except
 		}
 	}
 
+	/**
+	 * Tests {@link CheckpointBarrierUnaligner#processCancellationBarrier(CancelCheckpointMarker)}
+	 * abort the current pending checkpoint triggered by
+	 * {@link ThreadSafeUnaligner#notifyBarrierReceived(CheckpointBarrier, InputChannelInfo)}.
+	 */
+	@Test
+	public void testProcessCancellationBarrierWitchPendingCheckpoint() throws Exception {
```

Comment:
typo: witch -> with

Reply:
How about naming it `testProcessCancellationBarrierAfterNotifyBarrierReceived` and the other test `testProcessCancellationBarrierAfterProcessBarrier`?